### ISSUE TITLE
[PTR Onyxia] Boss pull + Onyxia Warders

### ISSUE THREAD

**Issue by Hyperbearr** [chromiecraft/chromiecraft#2957]
### What client do you play on?

enUS

### Faction

- [X] Alliance
- [ ] Horde

### Content Phase:

- [ ] Generic
- [ ] 1-19
- [ ] 20-29
- [ ] 30-39
- [ ] 40-49
- [ ] 50-59
- [X] 60

### Current Behaviour

When the boss is engaged in combat, previously killed warders remain dead

### Expected Blizzlike Behaviour

All Onyxia warders should respawn (but not engage) upon Onyxia being pulled so as to block exit from her room/lair.

### Steps to reproduce the problem

Kill all warders
Engage Onyxia

### AC rev. hash/commit

https://github.com/chromiecraft/azerothcore-wotlk/commit/6f20b9ece71984f1a11469e65ede8accda05653e

### Operating system

Ubuntu 20.04

### Modules

- [mod-ah-bot](https://github.com/azerothcore/mod-ah-bot)
- [mod-cfbg](https://github.com/azerothcore/mod-cfbg)
- [mod-chat-transmitter](https://github.com/azerothcore/mod-chat-transmitter)
- [mod-chromie-xp](https://github.com/azerothcore/mod-chromie-xp)
- [mod-desertion-warnings](https://github.com/azerothcore/mod-desertion-warnings)
- [mod-duel-reset](https://github.com/azerothcore/mod-duel-reset)
- [mod-eluna-lua-engine](https://github.com/azerothcore/mod-eluna-lua-engine)
- [mod-ip-tracker](https://github.com/azerothcore/mod-ip-tracker)
- [mod-low-level-arena](https://github.com/azerothcore/mod-low-level-arena)
- [mod-multi-client-check](https://github.com/azerothcore/mod-multi-client-check)
- [mod-pvp-titles](https://github.com/azerothcore/mod-pvp-titles)
- [mod-pvpstats-announcer](https://github.com/azerothcore/mod-pvpstats-announcer)
- [mod-queue-list-cache](https://github.com/azerothcore/mod-queue-list-cache)
- [mod-server-auto-shutdown](https://github.com/azerothcore/mod-server-auto-shutdown)
- [mod-save-inventory](https://github.com/Nyeriah/mod-save-inventory)
- [lua-carbon-copy](https://github.com/55Honey/Acore_CarbonCopy)
- [lua-exchange-npc](https://github.com/55Honey/Acore_ExchangeNpc)
- [lua-custom-worldboss](https://github.com/55Honey/Acore_CustomWorldboss)
- [lua-level-up-reward](https://github.com/55Honey/Acore_LevelUpReward)
- [lua-recruit-a-friend](https://github.com/55Honey/Acore_RecruitAFriend)
- [lua-send-and-bind](https://github.com/55Honey/Acore_SendAndBind)
- [lua-temp-announcements](https://github.com/55Honey/Acore_TempAnnouncements)
- [lua-zonecheck](https://github.com/55Honey/acore_Zonecheck)
- [lua-zone-debuff](https://github.com/55Honey/Acore_ZoneDebuff)

### Customizations

None

### Server

ChromieCraft

**Comment by Nyeriah** [comment]
I'm going to need sources on this one, it's not what happens on retail and you can reset her encounter at any time by just moving out of her chamber

**Comment by Hyperbearr** [comment]
> I'm going to need sources on this one, it's not what happens on retail and you can reset her encounter at any time by just moving out of her chamber

https://wowpedia.fandom.com/wiki/Onyxia_(Classic)

Thats the original wowpedia page about the fight. Ill dig for more sources tomorrow.

**Comment by Nyeriah** [comment]
I just tried this on retail, she spawns a single warder by the door whenever she is pulled. The mob also despawns when she resets. And yep she can be reset still regardless by just moving out of her room.

There's something odd about it though, because 1) the mob is so far back that you can move out of the room without pulling him 2) it'll just despawn along with her once you move out.

I suppose it's to prevent players from running into safe spots during fight

**Comment by Hyperbearr** [comment]
> I just tried this on retail, she spawns a single warder by the door whenever she is pulled. The mob also despawns when she resets. And yep she can be reset still regardless by just moving out of her room.
> 
> There's something odd about it though, because 1) the mob is so far back that you can move out of the room without pulling him 2) it'll just despawn along with her once you move out.
> 
> I suppose it's to prevent players from running into safe spots during fight

Yeah but retail is the WOTLK incarnation of the fight, so it depends what you're emulating here, I was under the impression the goal at the moment is to faithfully replicate the vanilla version of the fight rather than the WOTLK version of the fight?

**Comment by acidmanifesto** [comment]
Boss is pending a rewrite currently and will be reassessed after the rewrite.

**Comment by Rorschach91** [comment]
Fixed this july. I close this report as solved. In any case it can be re-opened when CC will be in Wotlk content.

**Comment by heyitsbench** [comment]
This hasn't been fixed.

**Comment by FrancescoBorzi** [comment]
> This hasn't been fixed.

@heyitsbench if you think this bug still exists can you please bring it to AC?
